market-contract/src/main.sw

```diff
@@ -146,16 +146,7 @@ impl Market for Contract {
     fn open_order(amount: u64, order_type: OrderType, price: u64) -> b256 {
         reentrancy_guard();
 
-        let asset_type = AssetType::Base;
-        open_order_internal(
-            amount,
-            asset_type,
-            order_type,
-            price,
-            storage
-                .matcher_fee
-                .read(),
-        )
+        open_order_internal(amount, order_type, price, storage.matcher_fee.read())
```

Comment:
Changed `open_order_internal` to not take `asset_type` as a input parameter since it is always AssetType::Base when opening an order.